rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/history/HistoryInstances.scala

```diff
@@ -203,22 +203,115 @@ object HistoryInstances {
       FlatMap[F].tailRecM((fullPath, trieNodesOnPath, none[Trie]))(go)
     }
 
-    def process(actions: List[HistoryAction]): F[History[F]] =
-      // TODO this is an intermediate step to reproduce all the trie behavior
-      // will evolve to a fold based implementation with partial tries
+    def findUncommonNode(
+        currentPath: KeyPath,
+        previousPath: KeyPath,
+        previousRoot: Trie
+    ): F[Option[Blake2b256Hash]] = {
+      def commonPrefixWithLastDiverging(path: KeyPath, affix: ByteVector): Boolean =
```

Comment:
It would make sense to move this and `pathTerminatesInAffix` to `KeyPathOps`

Reply:
https://rchain.atlassian.net/projects/RCHAIN/issues/RCHAIN-3531